src/payment/payment.service.ts

```diff
@@ -1,26 +1,56 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
 import { CreatePaymentDto } from './dto/create-payment.dto';
 import { UpdatePaymentDto } from './dto/update-payment.dto';
+import { PaymentInterface } from './interfaces/payment.interface';
 
 @Injectable()
 export class PaymentService {
-  create(createPaymentDto: CreatePaymentDto) {
-    return 'This action adds a new payment';
+  constructor(@InjectModel('Payment') private readonly paymentModel: Model<PaymentInterface>) {}
+
+  async create(createPaymentDto: CreatePaymentDto) {
+    const payment = await new this.paymentModel(createPaymentDto);
+    return payment.save();
+  }
+
+  async findAll() {
+    const payment = await this.paymentModel.find().exec();
+    if (!payment || !payment[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return payment;
   }
 
-  findAll() {
-    return `This action returns all payment`;
+  async findAllPaymentsForUser(userId: string) {
+    const user = await this.paymentModel.find({ userId }).exec();
+    if (!user || !user[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return user;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} payment`;
+  async findOne(id: string) {
+    const payment = await this.paymentModel.findOne({ _id: id }).exec();
+    if (!payment) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return payment;
   }
 
-  update(id: number, updatePaymentDto: UpdatePaymentDto) {
-    return `This action updates a #${id} payment`;
+  async update(id: number, updatePaymentDto: UpdatePaymentDto) {
+    const payment = await this.paymentModel.findByIdAndUpdate({ _id: id }, updatePaymentDto).exec();
+    if (!payment) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return payment;
```

Comment:
w zależności jakie body tutaj chcesz zwrócić , ale w tym momencie zwraca nie updatowanego paymenta, 
nie wiem czy przechodzą Ci testy, jesli tak to zostaw